Allow up to INT32_MAX max size in Array/Dictionary editor

JBrowne017 opened the pull request:
Quick update to editor_properties_array_dict max size. An array in the EditorInspector won't display a number higher than 1,000,000. The Range::Shared max value is stored in a double, as such the new max value has been updated to be DBL_MAX instead of an arbitrary magic number.

Resolves #77190

AThousandShips commented:
`DBL_MAX` is far larger than the maximum possible size for an array though

JBrowne017 commented on editor/editor_properties_array_dict.cpp:
```diff
@@ -311,7 +311,7 @@ void EditorPropertyArray::update_property() {
 			size_slider = memnew(EditorSpinSlider);
 			size_slider->set_step(1);
-			size_slider->set_max(1000000);
+			size_slider->set_max(DBL_MAX);
```
Yea I was doing more diving into the codebase and was looking at where it was actually used and it would be better to use a integer value instead. Would you still recommend INT_MAX over something like UINT32_MAX? I don't know a case where you would need a negative array size and there are a few spots where I see the size being cast to a unsigned 32 bit integer anyways.

AThousandShips replied:
It's a signed integer so not `UINT32_MAX`, the `resize` function takes `int` so don't think it's safe

JBrowne017 committed:
Replace magic num in editor_properties_array_dict

Quick update to editor/editor_properties_array_dict max size.
Currently, an array in the EditorInspector won't display a number higher than 1,000,000.
In place of the current magic number this sets the max to be the INT32_MAX.
This eludes the magic number in place and is sufficiently large.

Resolves #77190

AThousandShips commented:
You have duplicate and redundant commits, you will need to squash these, see [here](https://docs.godotengine.org/en/latest/contributing/workflow/pr_workflow.html#the-interactive-rebase)

AThousandShips reviewed (approved):
LGTM, some editor maintainer will have to give their opinion on if this is a good range but it makes sense to me and makes any exported array usable in the editor

JBrowne017 commented:
Yea I'm currently doing that. It's just been a minute since I've done a large rebase and squash in general. Thanks for the heads up though.

AThousandShips commented:
Remember to update the PR description and title

akien-mga commented:
Thanks! And congrats for your first merged Godot contribution :tada:

YuriSizov commented:
Cherry-picked for 4.0.4.
